pypsa/linopf.py

```diff
@@ -257,6 +261,74 @@ def define_unit_commitment_status_variables(n, sns, c):
     define_binaries(n, (sns, com_i), c, "status", mask=active)
 
 
+def define_generator_multiple_device(n):
+    c = "Generator"
+    p_nom_i = n.generators.index[n.generators.p_nom > 0]
+    ext_i = get_extendable_i(n, c)
+    if (ext_i.intersection(p_nom_i)).empty:
+        return
```

Comment:
the calculation of the indices may be done in a clear way probably.
I'd be prone adding a new parameter like "modular" with value True/False (default False) that specifies whether this functionality shall be implemented. However, this is not mandatory and we may use as flag the combination as "extendable" (that would be required anyway) and a non-zero p_nom.
In the following, I'll refer to the possible reformulation of `ext_i.intersection(p_nom_i)` as `mod_i`

regardless, on the choice, the indices mod_i may be calculated using query:
`n.df(c).query("extendable and (p_nom > 0)").index` or `n.df(c).query("extendable and modular").index`

@FabianHofmann or @fneum May you have preferences?
Personally, I'd be prone towards an additional option rather than overloading p_nom and extendable, but not sure what is your opinion.

Reply:
I'll replace it with: mod_i=n.df(c).query('p_nom_extendable and (p_nom>0)').index

If @FabianHofmann and @fneum have different preferences, we can change them accordingly :)